src/performance/gpu/amd/tdp.rs

```diff
@@ -221,29 +225,31 @@ impl TDP {
 }
 
 impl TDPDevice for TDP {
-
     fn tdp(&self) -> TDPResult<f64> {
         // Get the current stapm limit from ryzenadj
         match TDP::get_stapm_limit(&self) {
             Ok(result) => Ok(result.into()),
-            Err(err) => Err(TDPError::FailedOperation(err.to_string()))
+            Err(err) => Err(TDPError::FailedOperation(err.to_string())),
         }
     }
 
     fn set_tdp(&mut self, value: f64) -> TDPResult<()> {
         log::debug!("Setting TDP to: {}", value);
         if value < 1.0 {
             log::warn!("Cowardly refusing to set TDP less than 1W");
-            return Err(TDPError::InvalidArgument(format!("Cowardly refusing to set TDP less than 1W: provided {}W", value)));
+            return Err(TDPError::InvalidArgument(format!(
```

Comment:
Does the SteamOS kernel expose power1_cap_min and power1_cap_max in sysfs? Not blocking, something to look into as a way to auto detect limits.